tests/app/forms/field_handlers/test_date_handler.py

```diff
@@ -113,58 +115,53 @@ def test_get_referenced_offset_value_for_meta(app):
 @patch(
     "app.utilities.schema.load_schema_from_name", return_value=QuestionnaireSchema({})
 )
-def test_get_referenced_offset_value_for_answer_id(app):
+def test_get_referenced_offset_value_for_answer_id(app, value_source_resolver):
     answer_store = AnswerStore()
 
     test_answer_id = Answer(answer_id="date", value="2018-03-20")
     answer_store.add_or_update(test_answer_id)
-
+    value_source_resolver.answer_store = answer_store
     answer = {"maximum": {"value": {"identifier": "date", "source": "answers"}}}
 
-    handler = DateHandler(answer, answer_store=answer_store)
+    handler = DateHandler(answer, value_source_resolver)
     maximum_date = handler.get_date_value("maximum")
     maximum_date = handler.transform_date_by_offset(maximum_date, {"months": 1})
 
     assert maximum_date == convert_to_datetime("2018-04-20")
 
 
-# pylint: disable=unused-argument
-@patch(
-    "app.utilities.schema.load_schema_from_name", return_value=QuestionnaireSchema({})
-)
 @patch(
     "app.questionnaire.questionnaire_schema.QuestionnaireSchema.is_repeating_answer",
     return_value=True,
 )
-def test_get_referenced_offset_value_with_list_item_id(app, schema_mock):
+def test_get_referenced_offset_value_with_list_item_id(app, value_source_resolver):
     list_item_id = "abcde"
-    answer_store = AnswerStore()
 
     test_answer_id = Answer(
         answer_id="date", value="2018-03-20", list_item_id=list_item_id
     )
 
     location = Location(section_id="test", list_item_id=list_item_id)
 
-    answer_store.add_or_update(test_answer_id)
+    answer_store = AnswerStore()
 
-    answer = {
-        "maximum": {
-            "value": {"identifier": "date", "source": "answers"},
-            "offset_by": {"months": 1},
-        }
-    }
+    answer_store.add_or_update(test_answer_id)
+    value_source_resolver.answer_store = answer_store
+    value_source_resolver.location = location
+    value_source_resolver.list_item_id = list_item_id
+    answer = {"maximum": {"value": {"identifier": "date", "source": "answers"}}}
 
-    handler = DateHandler(answer, answer_store=answer_store, location=location)
+    handler = DateHandler(answer, value_source_resolver)
     maximum_date = handler.get_date_value("maximum")
+    maximum_date = handler.transform_date_by_offset(maximum_date, {"months": 1})
```

Comment:
Why does `transform_date_by_offset` need to be called rather than being part of the answer schema as it was previously?

Reply:
Resolved in https://github.com/ONSdigital/eq-questionnaire-runner/commit/fdc18d4747444751dbb2bea1da8180d1eea9d06e